test_rclcpp/CMakeLists.txt

```diff
@@ -131,6 +133,42 @@ if(AMENT_ENABLE_TESTING)
         "rclcpp")
     endif()
 
+    ament_add_gtest(local_parameters_cpp__${middleware_impl} "test/local_parameters.cpp")
+    add_dependencies(local_parameters_cpp__${middleware_impl} ${PROJECT_NAME})
```

Comment:
Before using the gtest target you must check that it actually exists (e.g. when gtest is not found):

    if(TARGET local_parameters_cpp__${middleware_impl})

Same below.